micropub.php

```diff
@@ -481,6 +496,37 @@ private static function handle_authorize_error( $code, $msg ) {
 		static::respond( $code, $msg );
 	}
 
+	private static function post_status( $mf2 ) {
+		$props = $mf2['properties'];
+		if ( isset( $props['visibility'] ) ) {
+			if ( array( 'private' ) === $props['visibility'] ) {
+				return 'private';
+			}
+			// Public or Private are the only options
+			if ( array( 'public' ) !== $props['visiblity'] ) {
+				return null;
+			}
+		}
+		if ( ! isset( $props['post-status'] ) ) {
+			return get_option( 'micropub_default_post_status' );
+		} else {
+			//  According to the proposed specification these are the only two properties supported.
+			// https://indieweb.org/Micropub-extensions#Post_Status
+			// For now these are the only two we will support even though WordPress defaults to 8 and allows custom
+			// But makes it easy to change
+			$statuslist = array( array( 'published' ), array( 'draft' ) );
+			if ( ! in_array( $props['post-status'], $statuslist, true ) ) {
+				return null;
+			}
+			// Map published to the WordPress property publish.
+			if ( array( 'published' ) === $props['post-status'] ) {
+				return 'publish';
+			}
+			return $props['post-status'];
+		}
+		return null;
```

Comment:
minor nit: probably remove this, for readability, since it will never actually be executed.

Reply:
...minor nit: that also means you could drop the `} else {` above and move its block up to top level.

Reply:
I have a test in the unit tests for someone sending in an unrecognized property. Won't it actually be executed if someone sends in an unrecognized error. Returning null will, when it is called return the unrecognized value error

Reply:
nope, this line will never run. the `if` and `else` paths above both end in `return`s. for clarity, let's replace it with an `// execution will never reach here` comment.